content/testing.md

```diff
@@ -0,0 +1,683 @@
+ 4. Starts up the test driver package ([see below](#driver-package)).
+
+What this means is that you can write tests in files with a certain filename pattern and know they'll not be included in normal builds of your app. When your app runs in test mode, those files will be loaded (and nothing else will), and they can import the modules you want to test. As we'll see this is ideal for [unit tests](#unit-testing) and [simple integration tests](#simple-integration-test).
+
+Additionally, Meteor offers a "full application" test mode. You can run this with `meteor test --full-app`.
+
+This is similar to test mode, with key differences:
+
+ 1. It loads test files matching `*.app-test[s].*` and `*.app-spec[s].*`.
+ 2. It **does** eagerly load our application code as Meteor normally would.
+
+This means that the entirety of your application (including for instance the web server and client side router) is loaded and will run as normal. This enables you to write much more [complex integration tests](#full-app-integration-test) and also load additional files for [acceptance tests](#acceptance-testing).
+
+Note that there is another test command in the Meteor tool; `meteor test-packages` is a way of testing Atmosphere packages, which we'll discuss in more detail in the [Writing Packages Article](writing-packages.html#testing).
+
+<h3 id="driver-packages">Driver packages</h3>
+
+When you run a `meteor test` command, you must provide a `--driver-package` argument. A test driver is a mini-application that runs in place of your app and runs each of your defined tests, whilst reporting the results in some kind of user interface.
+
+There are two main kinds of test driver packages:
+
+  - web-reporters which are Meteor applications and display a special test reporting web UI that you can view the test results in.
+
+  <img src="images/mocha-test-results.png">
+
+  - console-reporters that run completely on the command-line and are primary used for automated testing like [continuous integration](#ci) (as we'll see, typically PhantomJS is used to drive such tests).
+
+In this article, we'll use the popular [Mocha](https://mochajs.org) test runner alongside the [Chai](http://chaijs.com) assertion library to test our application. In order to write tests in Mocha, we can add the [`avital:mocha`](https://atmospherejs.com/avital/mocha) package to our app.
+
+```bash
+meteor add avital:mocha
+```
+
+This package also doesn't do anything in development or production mode (in fact it declares itself `testOnly` so it is not even included in those modes), but when our app is run in [test mode](#test-modes), it takes over, executing test code on both the client and server, and rendering results to the browser.
+
+Test files themselves (files named `*.[app]-test[s].*` or `*.[app]-spec[s].*`) can register themselves to be run by the test driver in the usual way for that testing library. For Mocha, that's by using `describe` and `it`:
```

Comment:
These regex patterns might be confusing for people who aren't familiar for that syntax. Perhaps, give an example file name.